packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -431,18 +432,39 @@ export class SCWSigner implements Signer {
     const state = store.getState();
     const subAccount = state.subAccount;
     if (subAccount?.address) {
-      const allAccounts = this.accounts.filter(
+      // Move the sub account to the top level accounts to make it active
+      const existingAccounts = this.accounts.filter(
         (account) => account.toLowerCase() !== subAccount.address.toLowerCase()
       );
-      this.accounts = allAccounts;
-
-      this.callback?.('accountsChanged', [subAccount.address, ...allAccounts]);
+      this.accounts = [subAccount.address, ...existingAccounts];
+      this.callback?.('accountsChanged', this.accounts);
       return subAccount;
     }
 
     // Wait for the popup to be loaded before sending the request
     await this.communicator.waitForPopupLoaded?.();
 
+    if (Array.isArray(request.params) && request.params.length > 0 && request.params[0].account) {
+      let keys: { type: string; publicKey: string }[];
+      if (request.params[0].account.keys && request.params[0].account.keys.length > 0) {
+        keys = request.params[0].account.keys;
+      } else {
+        const { account: ownerAccount } = await getCryptoKeyAccount();
+
+        if (!ownerAccount) {
+          throw standardErrors.provider.unauthorized('could not get subaccount owner account');
+        }
+
```

Comment:
should we only be doing this if type = create? with no keys being passed?

Reply:
i think it makes sense to add a check that type = create. i don't think it should only run if `enableAutoSubAccounts ` is true because this is behavior defined in the spec, independent of `enableAutoSubAccounts `